### PR TITLE
Feature/#278 3498.36 3498.72 favorites

### PR DESCRIPTION
**Pull request by saschahepp**
Code-Review PR

### CODE REVIEW

File: themes/sbvfrd/js/swissbib/Accordion.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  },
+
+  /**
+   * Saves state of expanded groups
+   *
+   */
+  saveExpandedGroups: function() {
+      var expandedGroupIds = [];
+
+      $("#accordion .in").each(function (index) {
```
Comment (saschahepp) [inline themes/sbvfrd/js/swissbib/Accordion.js]: $('#accordion').find('.in') is much faster

File: themes/sbvfrd/js/swissbib/Accordion.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          var currentItem = $(this);
+
+          //only saves state of first hierachie level
+          if (currentItem.attr('level') == 1) {
+              expandedGroupIds.push(currentItem.attr('id'));
+          }
+      });
+
+      if(expandedGroupIds.length > 0) {
+          var cookieData = {};
```
Comment (saschahepp) [inline themes/sbvfrd/js/swissbib/Accordion.js]: IMHO there's no need to save data by idRecord if you create a new object every time...

File: themes/sbvfrd/js/swissbib/Accordion.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            target.find('.in').collapse('hide');
+            swissbib.Accordion.saveExpandedGroups();
+        }
+    });
+
+
+    //on (re)load - open direct link library
+    var expandlib = swissbib.Accordion.getParameterByName('expandlib');
+    if (expandlib != null) {
+        $("#accordion #collapse-" + expandlib.split('-')[0]).collapse('show');
```
Comment (saschahepp) [inline themes/sbvfrd/js/swissbib/Accordion.js]: same as above, $('#accordion').find(... is faster and #accordion can be re-used

File: themes/sbvfrd/js/swissbib/Accordion.js (hunk 2 above)
Comment (drummer23) [inline themes/sbvfrd/js/swissbib/Accordion.js]: it is neccesary for recnozing if user has moved to a new page

File: themes/sbvfrd/js/swissbib/Accordion.js (hunk 3 above)
Comment (drummer23) [inline themes/sbvfrd/js/swissbib/Accordion.js]: done

File: themes/sbvfrd/js/swissbib/Accordion.js (hunk 1 above)
Comment (drummer23) [inline themes/sbvfrd/js/swissbib/Accordion.js]: done

**Pull request merged**